### PR TITLE
feat(compass-aggregation): adds drag-n-drop for use-cases available in sidepanel

### PR DESCRIPTION
**Pull request by himanshusinghs**
## Description
In this PR we bring the drag and drop behaviour for our side panel. With this change, users will be able to drag a use case from the side panel and drop it anywhere in the stage list to create a stage wizard. The "add wizard on click" functionality will still work.

A small preview:

https://github.com/mongodb-js/compass/assets/10037761/96ca6848-7b35-4380-9f61-c127f218472d

### Notes for reviewer:
Since the markup have been shifted around I would recommend the following:
- Start with the following since they are more related to mark up restructuring:
    - `pipeline-builder-workspace/index.tsx`
    - `pipeline-builder-workspace/pipeline-builder-as-text-workspace/index.tsx`
- Next head to `pipeline-builder-workspace/pipeline-builder-as-ui-workspace/index.tsx`, to understand the new markup together with DndContext and Resizable Side panel.
- To connect the dots look into the changes in `SortableList` and `AddStage`

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/index.tsx [hunk 1]
```diff
@@ -34,38 +40,45 @@ export const PipelineAsTextWorkspace: React.FunctionComponent<
 > = ({ isAutoPreview }) => {
   if (!isAutoPreview) {
     return (
+      <div className={workspaceContainerStyles}>
```
Comment (himanshusinghs) [inline packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/index.tsx:43]: Here, I have only nested the original markup inside the `div` with `workspaceContainerStyles`

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/index.tsx [hunk 2]
```diff
@@ -34,38 +40,45 @@ export const PipelineAsTextWorkspace: React.FunctionComponent<
...
+        >
+          <div className={noPreviewEditorStyles}>
+            <PipelineEditor />
+          </div>
+        </KeylineCard>
+      </div>
+    );
+  }
+  return (
+    <div className={workspaceContainerStyles}>
```
Comment (himanshusinghs) [inline packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/index.tsx:56]: Same here, nested the original markup inside the `div` with `workspaceContainerStyles`

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/index.tsx [hunk 3]
```diff
@@ -18,68 +15,26 @@ const containerStyles = css({
   height: '100%',
 });
 
-const workspaceStyles = css({
-  paddingBottom: spacing[3],
-  width: '100%',
-  overflow: 'auto',
-});
```
Comment (himanshusinghs) [inline packages/compass-aggregations/src/components/pipeline-builder-workspace/index.tsx:25]: This style that was critical to provide bottom padding, has been moved to individual workspaces

File: packages/compass-aggregations/src/components/add-stage/add-stage.tsx [hunk 4]
```diff
@@ -6,41 +6,89 @@ import {
...
 });
 
 type AddStageProps = {
+  index: number;
+  renderUseCaseDropMarker: boolean;
   variant: 'button' | 'icon';
   onAddStage: () => void;
 };
 
-export const AddStage = ({ onAddStage, variant }: AddStageProps) => {
```
Comment (mabaasit) [inline packages/compass-aggregations/src/components/add-stage/add-stage.tsx:35]: This component is not an AddStage anymore. Should we consider renaming it? Or add another component like `StageSeperator` or something, which can handle the logic of showing droppable area or `AddStage` button?

Comment (himanshusinghs) [reply packages/compass-aggregations/src/components/add-stage/add-stage.tsx:35]: That sounds good actually. I will take care of that. Thank you :)

File: packages/compass-aggregations/src/components/add-stage/add-stage.tsx [hunk 5]
```diff
@@ -9,33 +9,24 @@ import {
...
-const iconContainerStyles = css({
-  textAlign: 'center',
-  marginTop: spacing[1] / 2,
-  marginBottom: spacing[1] / 2,
-});
-
-const buttonContainerStyles = css({
-  textAlign: 'center',
-  marginTop: spacing[4],
-  marginBottom: spacing[3],
```
Comment (mabaasit) [inline packages/compass-aggregations/src/components/add-stage/add-stage.tsx:21]: we have reduced spacing at the bottom of link now. we should consider increasing it to maintain the previous UX.

**Review by mabaasit — APPROVED** [review]

**Pull request merged**